File: Selenium/Project/Java/Activity7.java
package project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.awt.*;
import java.awt.datatransfer.StringSelection;
import java.time.Duration;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Activity7 {
    WebDriver driver;
    @BeforeClass
    public void setUp()
    {
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/jobs/");
    }
    @Test
    public void verifyPostJob() throws InterruptedException {
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        Random rand = new Random();
        int randomNum = rand.nextInt((999 - 100) + 1) + 100;
        String emailId = "test"+randomNum+"@gmail.com";
        //Click on post a job
        driver.findElement(By
                .xpath("//a[contains(@href,'post-a')]")).click();
        //Enter required details
        driver.findElement(By.xpath("//input[@id='create_account_email']"))
                .sendKeys(emailId);
        driver.findElement(By.xpath("//input[@id='job_title']"))
                .sendKeys("Quality Analyst");
        WebElement jobType = driver.findElement(By.xpath("//select[@id='job_type']"));
        Select select = new Select(jobType);
        select.selectByVisibleText("Freelance");
        driver.switchTo().frame("job_description_ifr");

        WebElement descriptionBox = driver.findElement(By.xpath("//body[@id='tinymce']"));
        ((JavascriptExecutor)driver).executeScript("arguments[0].scrollIntoView(true);", descriptionBox);
        descriptionBox.click();
        String description = "This is a test job";
        Actions actions = new Actions(driver);
        actions.sendKeys(description).build().perform();
        driver.switchTo().defaultContent();
        driver.findElement(By
                .xpath("//input[@id='application']"))
                .sendKeys("testexample@gmail.com");
        driver.findElement(By
                .xpath("//input[@id='company_name']")).sendKeys("STest Job");
        driver.findElement(By.xpath("//input[@value='Preview']")).click();
        List<String> expected = new ArrayList<>();
        //expected.add(emailId);
        expected.add("Quality Analyst");
        expected.add("STest Job");
        expected.add("Freelance");
        //expected.add("This is a test job");
        //expected.add("testexample@gmail.com");

        //click on submit listing
        driver.findElement(By
                .xpath("//input[@value='Submit Listing']")).click();
        //Click on jobs
        driver.findElement(By
                .xpath("//a[contains(@href,'jobs/jobs')]")).click();
        //verify job listing
        By byFirstJobListing = By.xpath("(//ul[contains(@class,'listings')]" +
                        "//li//div//h3)[1]");
        wait.until(ExpectedConditions.visibilityOfElementLocated(byFirstJobListing));
        WebElement firstJobListing = driver.findElement(By
                .xpath("(//ul[contains(@class,'listings')]" +
                        "//li//div//h3)[1]"));
        //firstJobListing.click();
        System.out.println("Posted job is "+firstJobListing.getText());
        List<String> actual = new ArrayList<>();
        actual.add(firstJobListing.getText());
        actual.add(driver.findElement(By
                .xpath("(//ul[contains(@class,'listings')]" +
                        "//li//div//strong)[1]")).getText());
        actual.add(driver.findElement(By
                .xpath("(//ul[contains(@class,'listings')]" +
                        "//li//div//following-sibling::ul//li)[1]")).getText());
        System.out.println(actual);
        System.out.println(expected);
        Assert.assertEquals(actual, expected);
    }
    @AfterClass
    public void closeBrowser()
    {
        driver.close();
    }
}

File: Selenium/Project/Python/Activity7.py
import random
import time
import pytest
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait

@pytest.mark.usefixtures("setup")
class TestSite:

    def test_verify_post_job(self):
        wait = WebDriverWait(self.driver, 20)
        self.driver.get("https://alchemy.hguy.co/jobs/")
        randomNum = random.randint(100, 999)
        emailId = "test"+str(randomNum)+"@gmail.com"
        self.driver.find_element(By.XPATH, "//a[contains(@href,'post-a')]").click()
        self.driver.find_element(By.XPATH, "//input[@id='create_account_email']") \
            .send_keys(emailId)
        self.driver.find_element(By.XPATH, "//input[@id='job_title']") \
            .send_keys("Quality Analyst")
        jobType = self.driver.find_element(By.XPATH, "//select[@id='job_type']")
        select = Select(jobType)
        select.select_by_visible_text("Freelance")
        self.driver.switch_to.frame("job_description_ifr")
        descriptionBox = self.driver.find_element(By.XPATH, "//body[@id='tinymce']")
        self.driver.execute_script("arguments[0].scrollIntoView(true);", descriptionBox)
        descriptionBox.click()
        description = "This is a test job"
        actions = ActionChains(self.driver)
        actions.send_keys(description).perform()
        self.driver.switch_to.default_content()
        self.driver.find_element(By.XPATH, "//input[@id='application']") \
            .send_keys("testexample@gmail.com")
        self.driver.find_element(By.XPATH, "//input[@id='company_name']").send_keys("STest Job")
        self.driver.find_element(By.XPATH, "//input[@value='Preview']").click()
        expected = ["Quality Analyst", "STest Job", "Freelance"]
        time.sleep(5)
        wait.until(expected_conditions
               .visibility_of_element_located((By.XPATH,
                                               "//input[@value='Submit Listing']")))
        self.driver.find_element(By.XPATH, "//input[@value='Submit Listing']").click()
        self.driver.find_element(By.XPATH, "//a[contains(@href,'jobs/jobs')]").click()
        time.sleep(5)
        wait.until(expected_conditions
               .visibility_of_element_located((By.XPATH,
                                               "(//ul[contains(@class,'listings')]//li//div//h3)[1]")))
        firstJobListing = self.driver.find_element(By.XPATH, "(//ul[contains(@class,'listings')]//li//div//h3)[1]")
        print("Posted job is "+firstJobListing.text)
        actual = [firstJobListing.text,
              self.driver.find_element(By.XPATH, "(//ul[contains(@class,'listings')]//li//div//strong)[1]").text,
              self.driver.find_element(By.XPATH, "(//ul[contains(@class,'listings')]//li//div//following-sibling::ul//li)[1]").text]
        print(actual)
        print(expected)
        assert actual == expected



